### PR TITLE
technical term CLI

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Adds a technical term page for the CLI
- Adds the CLI page to the ToC

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

### Definition of Done
Please delete options that are not relevant.

- [X] I have made corresponding changes to the documentation

Thank you for submitting!

### CODE REVIEW

File: docs/terms/cli.md [hunk 1]
```diff
@@ -1,7 +1,94 @@
...
+## Overview
+
+### Definition
+
+CLI stands for Command Line Interface.
+
+### Features and promises
+
+The CLI provides useful convenience functions covering all the steps of working with Great Expectations.  CLI commands consist of a noun indicating what you want to operate on, and a verb indicating the operation to perform.  All CLI commands have help documentation that can be accessed by including the `--help` option.
+
```
Comment (cdkini) [inline docs/terms/cli.md]: Note that you can also get all these options by just running `great_expectations` without any args.

File: docs/terms/cli.md [hunk 2]
```diff
@@ -1,7 +1,94 @@
...
+To assist you in creating Expectation Suites, the CLI provides commands for listing available suites, creating new empty suites, creating new suites with scaffolding, editing existing suites, and deleting expectation suites.  You can read about these commands in the CLI with the command:
+
+```bash title="Terminal command"
+great_expectations suite --help
+```
+
+<ValidateHeader/>
+
+To assist you in Validating your data, the CLI provides commands for listing existing Checkpoints, running an existing Checkpoint, creating new Checkpoints, and creating python scripts that will run a Checkpoint.  You can read about these commands in the CLI with the command:
```
Comment (cdkini) [inline docs/terms/cli.md]: Could we capitalize Python please?

**Review by cdkini — APPROVED** [review]
LGTM!

File: docs/terms/cli.md (hunk 1 above)
Comment (Rachel-Reverie) [reply docs/terms/cli.md]: I added that as a way of getting a list of the available commands.

File: docs/terms/cli.md (hunk 2 above)
Comment (Rachel-Reverie) [reply docs/terms/cli.md]: Done.

**Pull request merged**